src/shared/sam/activation.ts

```diff
@@ -78,6 +78,13 @@ export async function activate(activateArguments: {
     )
 
     await detectSamCli(false)
+    activateArguments.extensionContext.subscriptions.push(
+        vscode.workspace.onDidChangeConfiguration(configurationChangeEvent => {
+            if (configurationChangeEvent.affectsConfiguration('aws.samcli.location')) {
+                detectSamCli(false)
```

Comment:
We might want to show the detection message in this case (`detectSamCli(true)`) since it's in response to a direct user action.

Reply:
This was tricky. The old logic showed different messages depending on whether autodetection was performed, but did not handle whether the user setting was actually changed. The latter must be handled in order to avoid redundant messages, because the `initialize()` function (and potentially other codepaths) set the user setting, which then triggers another `onDidChangeConfiguration` event...